src/ReverseProxy/SessionAffinity/ISessionAffinityPolicy.cs

```diff
@@ -28,6 +30,22 @@ public interface ISessionAffinityPolicy
     /// <returns><see cref="AffinityResult"/> carrying the found affinitized destinations if any and the <see cref="AffinityStatus"/>.</returns>
     AffinityResult FindAffinitizedDestinations(HttpContext context, ClusterState cluster, SessionAffinityConfig config, IReadOnlyList<DestinationState> destinations);
 
+    /// <summary>
+    /// Finds <see cref="DestinationState"/> to which the current request is affinitized by the affinity key.
+    /// </summary>
+    /// <param name="context">Current request's context.</param>
+    /// <param name="cluster">Current request's cluster.</param>
+    /// <param name="config">Affinity config.</param>
+    /// <param name="destinations"><see cref="DestinationState"/>s available for the request.</param>
+    /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
+    /// <returns><see cref="AffinityResult"/> carrying the found affinitized destinations if any and the <see cref="AffinityStatus"/>.</returns>
+    ValueTask<AffinityResult> FindAffinitizedDestinationsAsync(HttpContext context, ClusterState cluster, SessionAffinityConfig config, IReadOnlyList<DestinationState> destinations, CancellationToken cancellationToken = default)
```

Comment:
No need for defaults, we'll be the main caller and we want to make sure we remember the CT.
```suggestion
    ValueTask<AffinityResult> FindAffinitizedDestinationsAsync(HttpContext context, ClusterState cluster, SessionAffinityConfig config, IReadOnlyList<DestinationState> destinations, CancellationToken cancellationToken)
```